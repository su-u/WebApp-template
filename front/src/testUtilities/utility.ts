export const sel = (id: string) => `[data-test="${id}"]`;
