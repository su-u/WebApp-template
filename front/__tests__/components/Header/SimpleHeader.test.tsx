import React from 'react';
import { render } from 'enzyme';
import { MemoryRouter } from 'react-router';
import renderer from 'react-test-renderer';
import HeaderComponent from '../../../src/components/header/SimpleHeader';
import { sel } from '../../../src/testUtilities/utility';

describe('HeaderComponent', () => {
  describe('HeaderComponent', () => {
    let wrapper: any;

    beforeEach(() => {
      wrapper = render(
        <MemoryRouter initialEntries={['/']}>
          <HeaderComponent />
        </MemoryRouter>
      );
    });

    test('SnapShot', () => {
      const root = renderer.create(
        <MemoryRouter initialEntries={['/']}>
          <HeaderComponent />
        </MemoryRouter>
      );
      expect(root).toMatchSnapshot();
    });
    test('title', () => {
      expect(wrapper.find(sel('head-title')).text()).toEqual('テストアプリ');
    });
  });
});
