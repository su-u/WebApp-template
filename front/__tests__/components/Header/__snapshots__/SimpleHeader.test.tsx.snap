// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent HeaderComponent SnapShot 1`] = `
<div
  className="sc-AykKC POXrs"
>
  <div
    className="sc-AykKD hpsEDM"
  >
    <a
      data-test="head-title"
      href="/"
      onClick={[Function]}
    >
      テストアプリ
    </a>
  </div>
</div>
`;
